src/histogram.rs

```diff
@@ -25,17 +31,34 @@ impl HistogramExt for Histogram {
 /// (when it goes out of scope) or manually. Alternatively, it can be manually
 /// stopped and discarded in order to not record its value.
 pub struct HistogramTimer {
-    histogram: Histogram,
+    histogram: TimeHistogram,
     observed: bool,
     start: Instant,
 }
 
+#[derive(Debug)]
+struct Inner {
+    sum: AtomicU64,
+    count: AtomicU64,
+    buckets: Vec<(f64, AtomicU64)>,
+}
+
+impl HistogramExt for TimeHistogram {
```

Comment:
As we can now define `TimeHistogram`, we can remove `HistogramExt` altogether and make `start_timer` a method on `TimeHistogram`.